packages/contracts/contracts/Unirep.sol

```diff
@@ -283,6 +284,7 @@ contract Unirep is IUnirep, VerifySignature {
                 Polysum.add(epkPolysum, vals, EPK_R);
             }
             if (fieldIndex < sumFieldCount) {
+                if (change == 0) revert InvalidAttestation();
```

Comment:
We should allow 0 attestations. Practically this is a no-op, but there's nothing explicitly wrong with it, and the economics of the blockchain should naturally de-incentivize applications from doing it.

Reply:
I think if there is no limit in attestation it would be fine
but we have `MaxAttestations` limitation so people can send empty attestation and avoid receiving bad attestations or prevent others from receiving attestations.

Reply:
it only checks the sum field
so they can attest 0 to a replace field

Reply:
Yes, that's why I'm saying we don't need this check. It doesn't stop anything.

Reply:
but attest 0 to a replace field makes sense
like it can reset the replace field
but attest 0 to a sum field doesn't make change to the balance

Reply:
Yes it's a no-op, but it's not harmful, I don't think we shouldn't prevent applications from doing this.

Reply:
okay I think it is also fine not to handle this for attesters.